### PR TITLE
Added handling for Http 401/403 errors

### PR DESCRIPTION
**Pull request by slewis74**
Updated the work item link mapping code to explicitly check for Http 401/403 and return a specific error message about checking the access token.

Added a red/green E2E test around this scenario.

Relates to #94

### CODE REVIEW

File: source/JiraIntegration.sln.DotSettings [hunk 1]
```diff
@@ -1,3 +1,253 @@
 ﻿<wpf:ResourceDictionary xml:space="preserve" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" xmlns:s="clr-namespace:System;assembly=mscorlib" xmlns:ss="urn:shemas-jetbrains-com:settings-storage-xaml" xmlns:wpf="http://schemas.microsoft.com/winfx/2006/xaml/presentation">
+	<s:Boolean x:Key="/Default/CodeInspection/CodeAnnotations/NamespacesWithAnnotations/=Octopus_002ETools_002EAnnotations/@EntryIndexedValue">True</s:Boolean>
```
Comment (hnrkndrssn) [inline source/JiraIntegration.sln.DotSettings:2]: Was this on purpose?

**Review by hnrkndrssn — APPROVED** [review]
LGTM 👍

File: source/JiraIntegration.sln.DotSettings (hunk 1 above)
Comment (slewis74) [reply source/JiraIntegration.sln.DotSettings:2]: It was at the time, the formatting was wrong. I'll drop this out of here though and do a separate PR with this and all the resulting format updates.

**Comment by willsmith9182** [comment]
Legends, thank you. Much appreciated.

File: source/Server.E2E.Tests/WorkItemLinkMapperErrorHandlingTestScript.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+                log);
+
+            IResultFromExtension<WorkItemLink[]> result = null;
+            Should.NotThrow(
+                () =>
+                {
+                    result = workItemLinkMapper.Map(buildInformation);
+                });
+
+            result.ShouldBeOfType<FailureResultFromExtension<WorkItemLink[]>>("A failure should be received");
```
Comment (willsmith9182) [inline source/Server.E2E.Tests/WorkItemLinkMapperErrorHandlingTestScript.cs:50]: You'ave already merged this, but a handy thing from fluent assertions:
```
result.ShouldBeOfType<FailureResultFromExtension<WorkItemLink[]>>("A failure should be received")
.And.Subject.ErrorString.ShouldStartWith("Authentication failure, check the Jira access token is valid and has permissions to read work items");

```
`.And.Subject` chains the assertiosn but also subect is strongly typed based on the constraint used previously, as this is `OfType<T>` subject is the `T` from `OfType<T>`

You may also wanna look at assertion scopes when using multiple assertions :)

Comment (slewis74) [reply source/Server.E2E.Tests/WorkItemLinkMapperErrorHandlingTestScript.cs:50]: Nice, thanks for the tip, I'll definitely keep this in mind for future updates.

**Pull request merged**